### PR TITLE
cdo-analytics cookbook for running queries against Redshift

### PR DESCRIPTION
**Pull request by wjordan**
This PR adds a `cdo-analytics` cookbook that will install OS dependencies used for analytics queries against Redshift from `production-daemon` and/or `production-console`:

- `RedshiftClient` class for executing Redshift queries from Ruby scripts (e.g.: `RedshiftClient.instance.exec("SELECT * FROM ...")`;
- `bin/redshift` script for opening a `psql` console connected to the Redshift instance for running interactive queries;
- Install `sql-runner` for running playbooks of SQL scripts as needed.

The cookbook (and associated `pg` gem) will only be installed on the specific instances `production-daemon` and `production-console`, since we only have a Redshift instance in production and don't want to install these extra dependencies to other live frontends.

### CODE REVIEW

File: cookbooks/cdo-apps/Berksfile [hunk 1]
```diff
@@ -1,30 +1,12 @@
...
-  cloudwatch-extra-metrics
-  redis
-  solr
-  i18n
-).map {|x| "cdo-#{x}"}.
-  push('sudo-user').
-  each do |dep|
-  cookbook dep, path: "../#{dep}"
-end
+instance_eval File.read('../local_cookbooks.rb'), __FILE__
```
Comment (wjordan) [inline cookbooks/cdo-apps/Berksfile:4]: This is just a little bit of refactoring, will help maintain this long list of cookbook-dependencies moving forward.

File: Gemfile [hunk 2]
```diff
@@ -282,3 +282,13 @@ gem 'sort_alphabetical', github: 'grosser/sort_alphabetical'
...
+
+# Install pg gem only on specific production hosts.
+require_pg = -> do
+  require 'socket'
+  %w[production-daemon production-console].include?(Socket.gethostname)
+end
+
+install_if require_pg do
+  gem 'pg', require: false
+end
```
Comment (wjordan) [inline Gemfile:294]: Using [`install_if`](http://bundler.io/man/gemfile.5.html#INSTALL_IF) for this gem helps simplify the bundle/gemfile complexity we used as a workaround for this previously.

**Comment by bencodeorg** [comment]
Very excited!  I would primarily use it for executing sql snippets that create tables and insert records based on aggregates/combinations of other tables, not for the bulk retrieval of records. So not super worried about retrieval of sensitive data? There are plenty of existing ways to retrieve sensitive data (e.g., `bin/dashboard-sql`)

**Comment by bencodeorg** [comment]
Also @breville check out the `RedshiftClient` class here, might be useful if you're looking to query redshift to count projects in the future.

**Comment by ewjordan** [comment]
Late on this, but LGTM - I particularly like that this restricts installation to console and daemon, since that assuages any worry that we might end up leaning on Redshift for any user-initiated action. Good stuff!

**Pull request merged**